src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/AppInsights/AppInsightsLogger.cs

```diff
@@ -0,0 +1,312 @@
+                    default:
+                        if (value.Value is TimeSpan)
+                        {
+                            // if it's a TimeSpan, log the milliseconds
+                            metrics.Add(value.Key, ((TimeSpan)value.Value).TotalMilliseconds);
+                        }
+                        else if (value.Value is double || value.Value is int)
+                        {
+                            // try to insert the value as a metric
+                            metrics.Add(value.Key, Convert.ToDouble(value.Value));
+                        }
+
+                        // do nothing otherwise
+                        break;
+                }
+            }
+
+            _telemetryClient.TrackEvent(functionName + ".Aggregation", metrics: metrics);
+        }
+
+        private void LogFunctionResult(IEnumerable<KeyValuePair<string, object>> values, Exception exception)
+        {
+            IDictionary<string, object> scopeProps = AppInsightsScope.GetMergedStateDictionary() ?? new Dictionary<string, object>();
+
+            RequestTelemetry requestTelemetry = new RequestTelemetry();
+            requestTelemetry.Success = exception == null;
+            requestTelemetry.ResponseCode = "0";
+
+            // Set ip address to zeroes. If we find HttpRequest details below, we will update this
+            requestTelemetry.Context.Location.Ip = "0.0.0.0";
+
+            ApplyFunctionResultProperties(requestTelemetry, values);
+
+            // Functions attaches the HttpRequest, which allows us to log richer request details.
+            object request;
+            if (scopeProps.TryGetValue(ScopeKeys.HttpRequest, out request))
+            {
+                ApplyHttpRequestProperties(requestTelemetry, request as HttpRequestMessage);
+            }
+
+            // log associated exception details
+            if (exception != null)
+            {
+                ExceptionTelemetry exceptionTelemetry = new ExceptionTelemetry(exception);
+
+                string invocationId = values.Where(v => v.Key == LoggingKeys.InvocationId).Select(v => v.Value.ToString()).SingleOrDefault();
+                string functionName = values.Where(v => v.Key == LoggingKeys.Name).Select(v => v.Value.ToString()).SingleOrDefault();
+
+                exceptionTelemetry.Context.Operation.Id = invocationId;
+                exceptionTelemetry.Context.Operation.Name = functionName;
+
+                _telemetryClient.TrackException(exceptionTelemetry);
+            }
+
+            _telemetryClient.TrackRequest(requestTelemetry);
+        }
+
+        private static void ApplyHttpRequestProperties(RequestTelemetry requestTelemetry, HttpRequestMessage request)
+        {
+            if (request == null)
+            {
+                return;
+            }
+
+            requestTelemetry.Url = request.RequestUri;
+            requestTelemetry.Properties[LoggingKeys.HttpMethod] = request.Method.ToString();
```

Comment:
Not necessary. We typically put it into the `Name` like `GET /Home/Index`. Does it make sense for functions?